boot/bootutil/src/loader.c

```diff
@@ -909,9 +909,7 @@ boot_validated_swap_type(struct boot_loader_state *state,
 #if defined(PM_S1_ADDRESS) || defined(CONFIG_SOC_NRF5340_CPUAPP)
     const struct flash_area *secondary_fa =
         BOOT_IMG_AREA(state, BOOT_SECONDARY_SLOT);
-    struct image_header *hdr = (struct image_header *)secondary_fa->fa_off;
-    uint32_t vtable_addr = 0;
-    uint32_t *vtable = 0;
+    struct image_header *hdr = boot_img_hdr(state, BOOT_SECONDARY_SLOT);
```

Comment:
This seams to be OK as image headers were fetch already to the `state` structure before call to the boot_validate_swap_type() in lines:
https://github.com/nrfconnect/sdk-mcuboot/blob/17f6d045a10a8d2ba9f3dede820e9c6dd81a5c56/boot/bootutil/src/loader.c#L1827 and https://github.com/nrfconnect/sdk-mcuboot/blob/17f6d045a10a8d2ba9f3dede820e9c6dd81a5c56/boot/bootutil/src/loader.c#L1858